src/core/public/apm_system.ts

```diff
@@ -87,6 +74,45 @@ export class ApmSystem {
     });
   }
 
+  /* Hold the page load transaction open, until all resources actually finish loading */
+  private holdPageLoadTransaction(apm: ApmBase) {
+    this.pageLoadTransaction = apm.getCurrentTransaction();
+
+    // Keep the page load transaction open until all resources finished loading
+    if (this.pageLoadTransaction && this.pageLoadTransaction.type === 'page-load') {
+      // @ts-expect-error 2339
+      this.pageLoadTransaction.block(true);
+      this.pageLoadTransaction.mark('apmSetup');
+    }
+  }
+
+  /* Returns the number of resources that were loaded from network (vs. from disk cache) */
+  private getLoadedResourcesCount() {
+    try {
+      return window.performance
+        .getEntriesByType('resource')
+        .filter((r: Record<string, any>) => r.initiatorType === 'script' && r.transferSize > 0)
+        .length;
+    } catch (e) {
+      return -1;
```

Comment:
Why not `0/undefined/null`?

Reply:
I want to be able to detect error values
 - 0 is a ✅ value for `.length`
 - `addLabel` does not acccept `null` or `undefined` 

Therefor I thought that `-1` is a reasonable error value.